app/routes/playlists.js

```diff
@@ -0,0 +1,30 @@
+import SubRoute from 'backbone.subroute';
+import application from '../application';
+
+let PlaylistsRouter = SubRoute.extend({
+    routes: {
+        ':slug': 'show',
+        '': 'index'
+    },
+
+    index() {
+        console.log('indexPlaylist');
+        this.navigate('tracks', { trigger: true });
+    },
+
+    show(id) {
+        if (application.allPlaylists.length == 0) {
+            setTimeout(() => { // retry after the playlist has been loaded
```

Comment:
If this one is just to ensure the `allPlaylists` is fetched before running, I suggest you use the following workflow: as long as the `fetch` collection method wrap the `sync` method, it may be convenient to let `sync` returns a jqXHR object like the native Backbone.Sync version. This way, we can store the result of the `fetch` call (which is a Deferred object) on witch you can call a `done` promised method(https://api.jquery.com/jQuery.ajax/#jqXHR). You'll so be sure you're callback will be run when all Playlists are loaded, whatever time you run it :smiley:.